tests/performance/run_server_performance_test.py

```diff
@@ -0,0 +1,346 @@
+               'test in CSV format. The tasks are performed for all report '
+               'directories by all users.')
+
+    parser.add_argument('input',
+                        type=str,
+                        metavar='file/folder',
+                        nargs='+',
+                        default='~/.codechecker/reports',
+                        help="The analysis result files and/or folders.")
+    parser.add_argument('--url',
+                        type=str,
+                        metavar='PRODUCT_URL',
+                        dest='product_url',
+                        default='localhost:8001/Default',
+                        required=True,
+                        help="The URL of the product to store the results "
+                             "for, in the format of host:port/ProductName.")
+    parser.add_argument('-o', '--output',
+                        type=str,
+                        required=True,
+                        help="Output file name for printing statistics.")
+    parser.add_argument('-u', '--users',
+                        type=int,
+                        default=1,
+                        help="Number of users")
+    parser.add_argument('-t', '--time',
+                        type=int,
+                        default=-1,
+                        help="Timout in seconds. The script stops when the "
+                             "timeout expires. If a negative number is given "
+                             "then the script runs until it's interrupted.")
+    parser.add_argument('-b', '--beta',
+                        type=int,
+                        default=10,
+                        help="In the test users are waiting a random amount "
+                             "of seconds. The random numbers have exponential "
+                             "distribution of the beta parameter can be "
+                             "provided here.")
+    parser.add_argument('-v', '--verbose',
+                        action='store_true',
+                        help="Print the output of CodeChecker commands.")
+
+    return parser.parse_args()
+
+
+class StatManager:
+    """
+    This class stores the statistics of the single user events and prints them
+    in CSV format. To produce a nice output the users should do the same tasks
+    in the same order, e.g. they should all store, query and delete a run
+    in this order. In the output table a row belongs to each user. The columns
+    are the durations of the accomplished tasks.
+    """
+
+    def __init__(self):
+        # In this dictionary user ID is mapped to a list of key-value
+        # pairs: the key is a process name the value is its duration.
+        self._stats = defaultdict(list)
+
+    def add_duration(self, user_id, task_name, duration):
+        """
+        Add the duration of an event to the statistics.
+        """
+        self._stats[user_id].append((task_name, duration))
+
+    def print_stats(self, file_name):
```

Comment:
The output of this function looks like this:
```txt
User,Storage,Comparison,Reports,Delete,Storage,Comparison                        
1,0.110867,0.12792,0.126015,0.090653,0.093675,0.113359
```
For my eyes it would be better to write this statistics in the following format:
```txt
---==== Statistics ====---
--------------------------
Action     | Time (ms)
--------------------------
Storage    | 0.110867
Comparison | 0.12792
Reports    | 0.126015
Delete     | 0.090653
Storage    | 0.093675
Comparison | 0.113359
--------------------------
----==================----
Number of users: 2
----==================----
```
We use this at the CodeChecker parse command when we write the statistics to the output.